core/src/main/scala/io/projectglow/sql/SqlExtensionProvider.scala

```diff
@@ -100,12 +104,18 @@ object SqlExtensionProvider {
 
     functionRegistry.registerFunction(
       FunctionIdentifier("gq_summary_stats"),
-      exprs => ArrayStatsSummary.makeGqStats(exprs.head)
+      exprs => rewrite(GqSummaryStats(exprs.head))
     )
 
     functionRegistry.registerFunction(
       FunctionIdentifier("hard_calls"),
-      exprs => HardCalls(exprs(0), exprs(1), exprs(2), exprs.lift(3))
+      exprs => {
+        if (exprs.size == 4) {
```

Comment:
Nit: Should this be `exprs.size >= 4`?

Reply:
No, that should actually be an error. This will be handled in my followup PR, which performs much better parameter validation.